kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -107,7 +111,11 @@ class KyuubiSessionImpl(
         }
       try {
         _client = KyuubiSyncThriftClient.createClient(user, passwd, host, port, sessionConf)
-        _engineSessionHandle = _client.openSession(protocol, user, passwd, optimizedConf)
+        _engineSessionHandle = _client.openSession(
+          protocol,
+          user,
+          passwd,
+          optimizedConf ++ Map(KYUUBI_ENGINE_CREDENTIALS_KEY -> engineCredentials))
```

Comment:
I think a more safer way is that check `engineCredentials` is vaildated before inject into engine, so if engine get the KYUUBI_ENGINE_CREDENTIALS_KEY that means it must has credentials.